cns/networkcontainers/networkcontainers.go

```diff
@@ -57,3 +88,104 @@ func (cn *NetworkContainers) Delete(networkContainerID string) error {
 	log.Printf("[Azure CNS] NetworkContainers.Delete finished.")
 	return err
 }
+
+// This function gets the flattened network configuration (compliant with azure cni) in bytes array format
+func getNetworkConfig(configFilePath string) ([]byte, error) {
+	content, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	var configMap map[string]interface{}
+	if err = json.Unmarshal(content, &configMap); err != nil {
+		log.Printf("[Azure CNS] Failed to unmarshal network configuration with error %v", err)
+		return nil, err
+	}
+
+	// Get the plugins section
+	pluginsSection := configMap[PluginsStr].([]interface{})
+	flatNetConfigMap := pluginsSection[0].(map[string]interface{})
+
+	if flatNetConfigMap == nil {
+		msg := "[Azure CNS] " + PluginsStr + " section of the network configuration cannot be empty."
+		log.Printf(msg)
+		return nil, errors.New(msg)
+	}
+
+	// insert version and name fields
+	flatNetConfigMap[VersionStr] = configMap[VersionStr].(string)
+	flatNetConfigMap[NameStr] = configMap[NameStr].(string)
+
+	// convert into bytes format
+	netConfig, err := json.Marshal(flatNetConfigMap)
+	if err != nil {
+		log.Printf("[Azure CNS] Failed to marshal flat network configuration with error %v", err)
+		return nil, err
+	}
+
+	return netConfig, nil
+}
+
+func args(action, path string, rt *libcni.RuntimeConf) *invoke.Args {
+	return &invoke.Args{
+		Command:     action,
+		ContainerID: rt.ContainerID,
+		NetNS:       rt.NetNS,
+		PluginArgs:  rt.Args,
+		IfName:      rt.IfName,
+		Path:        path,
+	}
+}
+
+func pluginErr(err error, output []byte) error {
+	if err != nil {
+		if _, ok := err.(*exec.ExitError); ok {
+			emsg := types.Error{}
+			if err := json.Unmarshal(output, &emsg); err != nil {
+				emsg.Msg = fmt.Sprintf("netplugin failed but error parsing its diagnostic message %s: %+v", string(output), err)
+			}
+
+			return &emsg
+		}
+	}
+
+	return err
+}
+
+func execPlugin(rt *libcni.RuntimeConf, netconf []byte, operation, path string) error {
+	switch operation {
+	case "ADD":
```

Comment:
doesn't case statements need to be indented?

Reply:
not in go style